python/cudf/cudf/tests/test_df_protocol.py

```diff
@@ -7,25 +7,54 @@
 import pytest
 
 import cudf
-from cudf.core.buffer import Buffer
-from cudf.core.column import build_column
+from cudf.core._compat import PANDAS_GE_150
+from cudf.core.column import as_column, build_column
 from cudf.core.df_protocol import (
     DataFrameObject,
     _CuDFBuffer,
     _CuDFColumn,
     _DtypeKind,
-    _from_dataframe,
+    _MaskKind,
+    _protocol_buffer_to_cudf_buffer,
+    from_dataframe,
     protocol_dtype_to_cupy_dtype,
 )
 from cudf.testing._utils import assert_eq
 
 
+@pytest.fixture
+def pandas_df():
+    return pd.DataFrame({"a": [1, 2, 3], "b": ["x", "y", None]})
+
+
+def assert_validity_equal(protocol_buffer, cudf_buffer, size, null, valid):
+    if null == _MaskKind.BYTEMASK:
+        protocol_mask = _protocol_buffer_to_cudf_buffer(protocol_buffer)
+        assert_eq(
+            as_column(protocol_mask, dtype="bool"),
+            as_column(cudf_buffer, dtype="bool"),
+        )
+    elif null == _MaskKind.BITMASK:
+        protocol_mask = _protocol_buffer_to_cudf_buffer(protocol_buffer)
```

Comment:
Comparing bitmasks is harder because we don't have utilities in Python to compare two bitmasks AFAICT. Instead, we construct two string columns composed entirely of empty strings, assign them the "left" and "right" bitmasks respectively, and then use `assert_eq` to compare the resulting columns.

Open to other suggestions here :-)

Reply:
If we construct a column of empty strings, it still requires offsets to be allocated/generated (which are `size_type = int32`). A more lightweight approach would be to use an `int8` data type with this mask.

Reply:
Fixed to use an int8 column here instead